pom.xml

```diff
@@ -392,26 +392,65 @@
 				</plugins>
 			</build>
 		</profile>
-		<!-- Error prone; plugin is recommended but only works with Java 9 and newer.
-		Remove the two profiles and pull the pluginManagement contents up into the main config when the project moves off Java 8. -->
+
+		<!-- JDK 16+ configuration -->
+		<profile>
+			<id>jdk_16_and_up</id>
+			<activation>
+				<jdk>[16,]</jdk>
+			</activation>
+			<build>
+				<pluginManagement>
+					<plugins>
+						<plugin>
+							<groupId>org.apache.maven.plugins</groupId>
+							<artifactId>maven-surefire-plugin</artifactId>
+							<configuration>
+								<!-- illegal-access=permit is needed to make GCLIB proxying work, used in Datastore -->
+								<argLine>-Xms512m -Xmx512m --illegal-access=permit</argLine>
+								<skip>${skip.surefire.tests}</skip>
+								<excludes>
+									<exclude>${integration-test.pattern}</exclude>
+								</excludes>
+							</configuration>
+						</plugin>
+					</plugins>
+				</pluginManagement>
+			</build>
+		</profile>
+		<!-- Error prone; JDK 8 requires a different configuration than 9+ -->
 		<profile>
 			<id>errorprone_java_9_and_up</id>
 			<activation>
-				<jdk>[9,)</jdk>
+				<jdk>[9,]</jdk>
```

Comment:
Should this be 9 through 15, or 9 through latest? As is, both `errorprone_java_9_and_up` and `jdk_16_and_up` will fire on 16.

Reply:
Good point. I was just blindly backporting from the new repo.

Reply:
Actually, I think this is intentional. The two profiles combine for 16.